antarest/study/business/link_management.py

```diff
@@ -52,38 +174,81 @@ class LinkManager:
+                    "display_comments": link_properties.get("display-comments"),
+                    "filter_synthesis": link_properties.get("filter-synthesis"),
+                    "filter_year_by_year": link_properties.get("filter-year-by-year"),
+                }
+                ui_parameters = {}
                 if with_ui and links_config and link in links_config:
-                    ui_info = LinkUIDTO(
-                        color=f"{links_config[link].get('colorr', '163')},{links_config[link].get('colorg', '163')},{links_config[link].get('colorb', '163')}",
-                        width=links_config[link].get("link-width", 1),
-                        style=links_config[link].get("link-style", "plain"),
+                    ui_parameters.update(
+                        {
+                            "colorr": str(link_properties.get("colorr", DEFAULT_COLOR)),
+                            "colorb": str(link_properties.get("colorb", DEFAULT_COLOR)),
+                            "colorg": str(link_properties.get("colorg", DEFAULT_COLOR)),
+                            "link_width": link_properties.get("link-width", 1.0),
+                            "link_style": link_properties.get("link-style", LinkStyle.PLAIN),
+                        }
+                    )
+                    link_creation_data.update(ui_parameters)
+                else:
+                    link_creation_data.update(
+                        {"colorr": None, "colorb": None, "colorg": None, "link_width": None, "link_style": None}
                     )
-                result.append(LinkInfoDTO(area1=area_id, area2=link, ui=ui_info))
+                link_info_dto = LinkInfoFactory.create_link_info(**link_creation_data)
 
+                result.append(link_info_dto)
         return result
 
-    def create_link(self, study: RawStudy, link_creation_info: LinkInfoDTO) -> LinkInfoDTO:
+    def create_link(self, study: RawStudy, link_creation_info: LinkInfoDTOType) -> LinkInfoDTOType:
+        study_version = int(study.version)
+        link_info_dto = LinkInfoFactory.create_link_info(
+            version=int(study.version),
+            area1=link_creation_info.area1,
+            area2=link_creation_info.area2,
+            hurdles_cost=link_creation_info.hurdles_cost,
+            loop_flow=link_creation_info.loop_flow,
+            use_phase_shifter=link_creation_info.use_phase_shifter,
+            transmission_capacities=link_creation_info.transmission_capacities,
+            asset_type=link_creation_info.asset_type,
+            display_comments=link_creation_info.display_comments,
+            colorr=link_creation_info.colorr,
+            colorb=link_creation_info.colorb,
+            colorg=link_creation_info.colorg,
+            link_width=link_creation_info.link_width,
+            link_style=link_creation_info.link_style,
+            filter_synthesis=link_creation_info.filter_synthesis if study_version >= 820 else None,
```

Comment:
As you didn't cast link_creation_dto as a LinkInfoDTO820 instance, mypy doesn't know you can ask for this field. 
Little workaround to fix the issue: 
```python
filter_synthesis = None
filter_year_by_year = None
if study_version >= 820:
    link_creation_info = t.cast(LinkInfoDTO820, link_creation_info)
    filter_synthesis = link_creation_info.filter_synthesis
    filter_year_by_year = link_creation_info.filter_year_by_year
```